asm_to_pil/src/lib.rs

```diff
@@ -206,21 +105,31 @@ impl<'a, T: FieldElement> ASMPILConverter<'a, T> {
             self.registers
                 .iter()
                 .filter_map(|(name, reg)| {
-                    reg.update_expression().map(|mut update| {
-                        if Some(name) == self.pc_name.as_ref() {
+                    reg.update_expression().map(|rhs| {
+                        let lhs = next_reference(name);
+                        use RegisterTy::*;
+                        let (lhs, rhs) = match reg.ty {
                             // Force pc to zero on first row.
-                            update = build_mul(
-                                build_sub(build_number(1u64), next_reference("first_step")),
-                                update,
-                            )
-                        }
+                            Pc => (
+                                lhs,
+                                build_mul(
+                                    build_sub(build_number(1u64), next_reference("first_step")),
+                                    rhs,
+                                ),
+                            ),
+                            // Unconstrain read-only registers when calling `_reset`
+                            ReadOnly => {
+                                let not_reset =
+                                    build_sub(build_number(1u64), direct_reference("instr__reset"));
```

Comment:
same thing here, using a hardcoded constant that is defined in a different crate

Reply:
This is fine, as `first_step` is declared here. I changed the others.